mcp23017.c

```diff
@@ -65,36 +66,39 @@ static void updateRegisterBit(i2c *i2c, uint8_t pin, uint8_t pValue, uint8_t por
 i2c* mcp23xx_init(uint8_t i2c_bus, uint8_t i2c_address) {
 	i2c *i2c = libsoc_i2c_init(i2c_bus, i2c_address);
 
-	if (i2c == NULL) {
-		perror("libsoc_i2c_init failed");
-		return i2c;
+	if (i2c == NULL || mcp_begin(i2c) != EXIT_SUCCESS) {
```

Comment:
Can we split this check. This will return NULL even if mcp_begin fails.

Reply:
what should this func return if mcp_begin fails?